qiskit_optimization/algorithms/optimization_algorithm.py

```diff
@@ -566,7 +568,25 @@ def generate_solution(bitstr, qubo, probability):
             )
 
         solutions = []
-        if isinstance(eigenvector, dict):
+        if isinstance(eigenvector, QuasiDistribution):
```

Comment:
I guess here it doesn't matter that we might have negative values, which I think might happen for an error mitigated result. Such values will be discarded anyway by the threshold. Maybe we don't need to bother about the converting to a nearest distribution - which though might change the result from here I guess.

Reply:
`OptimizationAlgorithm` does not have any option to control `_eigenvector_to_solutions` method. So, we need to introduce an option, e.g., `nearest_probability`, to the constructor of `OptimizationAlgorithm`. What do you think?

Reply:
This PR is already a bit too big. So, I want to leave this option for another PR. Is it OK?

Reply:
Sure, a separate PR to address this sounds fine. Gives a bit more time too to think/consider options.